curvefs/src/metaserver/copyset/copyset_node.cpp

```diff
@@ -351,10 +352,12 @@ int CopysetNode::on_snapshot_load(braft::SnapshotReader* reader) {
     LOG(INFO) << "Copyset " << name_ << " begin to load snapshot from '"
               << reader->get_path() << "'";
 
+    isLoading_.store(true, std::memory_order_release);
```

Comment:
```cpp
struct CopysetLoadingGuard {
    CopysetLoadingGuard(std::atomic& flag) : flag(flag) {
           flag.store(true, ...);
    }

    ~CopysetLoadingGuard() {
          flags.stroe(false, ....);
    }


    std::atomic& flags;
};
```

in this way, you don't have to manually set `isLoading_` to false in every return point.

Reply:
agree

Reply:
done